server/services/src/test/java/org/zanata/service/impl/TranslationFileServiceImplTest.java

```diff
@@ -20,31 +20,153 @@
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Mockito.when;
 
 @RunWith(CdiUnitRunner.class)
 public class TranslationFileServiceImplTest extends ZanataTest {
 
     @Inject
     TranslationFileServiceImpl transFileService;
 
+    @Mock
+    @Produces
+    private ProjectIterationDAO projectIterationDAO;
+
+    @Mock
+    @Produces
+    private DocumentDAO documentDAO;
+
     @Produces @Mock Session session;
 
     @Test
     @InRequestScope
     public void hasPlainTextAdapter() {
         assertThat(transFileService.hasAdapterFor(DocumentType.PLAIN_TEXT)).isTrue();
     }
+
+    @Test
+    @InRequestScope
+    public void parsePoInFileProject() throws Exception {
+        String project = "test";
+        String version = "master";
+        String docId = "test.pot";
+        String poContent = "msgid \"\"\n" +
+                "msgstr \"\"\n" +
+                "\"Project-Id-Version: test-master\\n\"\n" +
+                "\n" +
+                "#, c-format\n" +
+                "msgid \"Thing 1\"\n" +
+                "msgid_plural \"%d files removed\"\n" +
+                "msgstr[0] \"1 aoeuaouaou\"\n" +
+                "msgstr[1] \"%d aoeuaouao\"\n" +
+                "msgstr[2] \"\"";
+        File tempFile = File.createTempFile("test", ".po");
+        Files.write(tempFile.toPath(), poContent.getBytes());
+        InputStream stream = new FileInputStream(tempFile);
+
+        HProject hProject = new HProject();
+        hProject.setDefaultProjectType(ProjectType.File);
+        HProjectIteration hProjectIteration = new HProjectIteration();
+        hProjectIteration.setProject(hProject);
+        hProjectIteration.setProjectType(ProjectType.File);
+        HDocument hDocument = new HDocument();
+
+        when(projectIterationDAO.getBySlug(project, version))
+                .thenReturn(hProjectIteration);
+        when(documentDAO.getByDocIdAndIteration(hProjectIteration, docId))
+                .thenReturn(hDocument);
+
+        TranslationsResource translationsResource = transFileService
+                .parseTranslationFile(stream, tempFile.getName(),
+                "ru", project, version, docId, Optional.absent());
+        assertThat(translationsResource.getTextFlowTargets().get(0)
+                .getContents().get(0)).isEqualTo("1 aoeuaouaou");
+    }
+
+    @Test
+    @InRequestScope
+    public void parseAdapterFileInFileProject() throws Exception {
+        String project = "test";
+        String version = "master";
+        String docId = "test.properties";
+        String plaintextContent = "first: test message";
+        File tempFile = File.createTempFile("test", ".properties");
+        Files.write(tempFile.toPath(), plaintextContent.getBytes());
+        InputStream stream = new FileInputStream(tempFile);
```

Comment:
You forgot to close it. But a ByteArrayInputStream would avoid the need for a temp file.

Reply:
Done